source/unix/openssl_aes.c

```diff
@@ -14,8 +14,21 @@ struct openssl_aes_cipher {
     struct aws_byte_buf working_buffer;
 };
 
+static struct aws_byte_cursor s_empty_plain_text = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("");
+
 static int s_encrypt(struct aws_symmetric_cipher *cipher, struct aws_byte_cursor input, struct aws_byte_buf *out) {
 
+    /*
+     * Openssl 1.1.1 and its derivatives like aws-lc and boringssl do not handle
+     * the case of null input of 0 len gracefully in update (it succeeds, but
+     * finalize after it will fail). Openssl 3.0 fixed this. Other crypto implementations
+     * do not have similar issue.
+     * To workaround the issue, replace null cursor with empty cursor.
+     */
+    if (input.ptr == NULL && input.len == 0) {
```

Comment:
Don't know the context much, but should we handle the case where the length is 0 and ptr is pointed to garbage? If yes, then
```suggestion
    if (input.len == 0) {
```

Reply:
its kinda grey area. i though we didnt consider cursor with garbage ptr and 0 len a valid cursor. but looks like anything with 0 len is valid. it does not seem to matter much as openssl can handle anything non-null, but sure for simplicity we can remove null check from this if